### PR TITLE
feat: add ability to set raw storage slots [APE-1088]

### PR DESCRIPTION
**Pull request by fubuloubu**
### What I did

Adds a mock API (to be implemented by testing providers) for setting raw storage slots

fixes: #360

### Checklist

- [x] All changes are completed
~~- [ ] New test cases have been added~~
- [x] Documentation has been updated

### CODE REVIEW

File: src/ape/api/providers.py [hunk 1]
```diff
@@ -508,6 +508,19 @@ def set_code(  # type: ignore[empty-body]
             code (:class:`~ape.types.ContractCode`): The new bytecode.
         """
 
+    @raises_not_implemented
+    def set_storage(  # type: ignore[empty-body]
+        self, address: AddressType, slot: int, value: HexBytes
```
Comment (antazoey) [inline src/ape/api/providers.py:513]: I thought I had remembered a virtual machine that didn't use storage slots, maybe it was Cairo?
Just wondering if this method would have a separate meaning then, if we should rename `slot`, stuff like that.

**Review by antazoey — APPROVED** [review]
looks good but wondering about systems without using storage slots

File: src/ape/api/providers.py (hunk 1 above)
Comment (fubuloubu) [reply src/ape/api/providers.py:513]: Could just call it `key` I suppose, most VMs come with some sort of persistent key-value store for a stateful blockchain

**Comment by fubuloubu** [comment]
> questions: does eth-tester or geth have implementations of this?

eth-tester doesn't, but you can probably directly set the storage database through the py-evm backend:
https://github.com/ethereum/py-evm/blob/eaa35b2138a65f5822780e514834e5222f7a42cf/eth/vm/logic/storage.py#L66

geth I don't think supports this

**Comment by antazoey** [comment]
> eth-tester doesn't, but you can probably directly set the storage database through the py-evm backend: 

maybe we could convince the Snake Charmers to implement this in `EthTester`

**Pull request merged**